### PR TITLE
Get rid of underscores in IRectangle (resolve korlibs/korma#45)

### LINKED ISSUE #45: Get rid of underscores in IRectangle
Current state seems to have redundant properties:

```kotlin
interface IRectangle {
    val _x: Double
    val _y: Double
    val _width: Double
    val _height: Double
}
val IRectangle.x get() = _x
val IRectangle.y get() = _y
val IRectangle.width get() = _width
val IRectangle.height get() = _height
```

I think it can become like this:
```kotlin
interface IRectangle {
    val x: Double
    val y: Double
    val width: Double
    val height: Double
}
```

As [discussed](https://discord.com/channels/728582275884908604/728616376910217258/794909014184951828), it can be handy but low-performant on native.

### PR DESCRIPTION
**Pull request by SerVB**
This resolves korlibs/korma#45.

### CODE REVIEW

File: korma/src/commonMain/kotlin/com/soywiz/korma/geom/Rectangle.kt [hunk 1]
```diff
@@ -4,10 +4,10 @@ import com.soywiz.korma.internal.*
 import com.soywiz.korma.interpolation.*
 
 interface IRectangle {
```
Comment (soywiz) [inline korma/src/commonMain/kotlin/com/soywiz/korma/geom/Rectangle.kt:6]: Maybe you can extend IPoint now?

**Review by soywiz — COMMENTED** [review]
Nice. I think if this is not optimal right now because devirtualization is not working, this should be handled at Kotlin/Native level if possible.

Could you add `IRectangle._x`, etc. with `@Deprecated+replace with`. Not sure if someone is using it. Hopefully only used internally, but let's give it a couple of versions before removing them completely just in case

File: korma/src/commonMain/kotlin/com/soywiz/korma/geom/Rectangle.kt (hunk 1 above)
Comment (SerVB) [reply korma/src/commonMain/kotlin/com/soywiz/korma/geom/Rectangle.kt:6]: Seems you are talking about korlibs/korma#44. Yep, I'm planning to do it in another PR, after we deal with this one.

**Comment by soywiz** [comment]
What did you think about providing deprecated extensions (maybe hidden) `_x`, `_y`, `_width` and `_height` in the end?

**Comment by SerVB** [comment]
Sorry, I don't have much time now. I will do it of course, I'll add a commit to this PR, hopefully on this weekend, and ping you.

Actually, I've reacted with 👀 to note that I've seen your message and will proceed a bit later 😉

**Comment by soywiz** [comment]
Sure, no rush!

**Comment by SerVB** [comment]
@soywiz, I've used ERROR deprecation level; I think HIDDEN makes message and ReplaceWith useless, so I've filed a ticket here: https://youtrack.jetbrains.com/issue/KT-45443

**Comment by soywiz** [comment]
Thanks @SerVB !

B8EE1FEC-C16A-4AF8-9003-D4B84D3A9B62

**Pull request merged**